GoogleForCreators/web-stories-wp
# Right Click Menu: Clear multiple element styles

## Context

Separated from user feature request #4616 to make sure initial new menus for when multiple elements are selected has isolated test time.

Branch from #9832. Once that is merged this will be smaller.

## Summary

Adds functionality to clear styles for multiple elements at the same time using the context menu.

## Relevant Technical Choices

Elements that couldn't be reset (text) still may not be reset even if selected.

## To-do

n/a

## User-facing changes


![clear-styles](https://user-images.githubusercontent.com/22185279/144838020-073e4259-c212-4546-8962-efbaedab11f2.gif)

## Testing Instructions






This PR can be tested by following these steps:

1. Add multiple elements to the canvas of different types. Change their styles.
2. Select them all using the `Shift` key
3. Right click on any of the selected elements or the frame
4. Click `Clear Styles`. Should see any image and shape elements get reset to defaults. All text elements should not be changed.
5. Click `Undo` in the snackbar that shows up
6. Should see element styles all revert to what they originally were.
7. `Undo` should work with the keyboard shorcut `mod + z`


## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #9885
